tests/test_batch_select.ts

```diff
@@ -0,0 +1,177 @@
+import Vue from "vue";
+import Component from "vue-class-component";
+
+import { Wrapper } from '@vue/test-utils';
+
+import _ from 'lodash';
+import * as sinon from 'sinon';
+
+import BatchSelect from "@/components/batch_select.vue";
+
+import { managed_mount } from '@/tests/setup';
+import { emitted } from '@/tests/utils';
+
+
+interface TestObj {
+    value: string;
+}
+
+let obj1: TestObj = {
+    value: "objectA-a"
+};
+let obj2: TestObj = {
+    value: "objectA-b"
+};
+let obj3: TestObj = {
+    value: "otherObject"
+};
+
+let objects: TestObj[] = [obj1, obj2, obj3];
+let selected: TestObj[] = [obj1];
+
+@Component({
+    template: `<div>
+                 <batch-select
+                     v-model="selected"
+                     :choices="objects"
+                     :are_items_equal="are_items_equal"
+                     :filter_fn="filter_fn"
+                     @items_selected="items_selected($event)"
+                     v-slot="{ item }"
+                     ref="batch_select"
+                 >
+                     {{ item.value }}
+                 </batch-select>
+                 <div class="change-selected"
+                      @click="change_selected">
+                 </div>
+               </div>`,
+    components: {
+        'batch-select': BatchSelect
+    }
+})
+class WrapperComponent extends Vue {
+    objects: TestObj[] = objects;
+    selected: TestObj[] = selected;
+
+    filter_fn(obj: TestObj, filter_text: string) {
+        return _.includes(obj.value, filter_text);
+    }
+
+    are_items_equal(lhs: TestObj, rhs: TestObj) {
+        return lhs.value === rhs.value;
+    }
+
+    items_selected(object: TestObj) {}
+
+    change_selected() {
+        this.selected = [obj3];
+    }
+}
+
+describe('BatchSelect', () => {
+    let wrapper: Wrapper<WrapperComponent>;
+    let batch_select_wrapper: Wrapper<BatchSelect>;
+    let items_selected_spy: sinon.SinonSpy;
+
+    beforeEach(() => {
+        wrapper = managed_mount(WrapperComponent);
+        items_selected_spy = sinon.spy(wrapper.vm, 'items_selected');
+        batch_select_wrapper = wrapper.findComponent({ref: 'batch_select'}) as Wrapper<BatchSelect>;
+    });
+
+    test('sets d_selected_items to a deep copy of value', async () => {
+        expect(batch_select_wrapper.vm.d_selected_items).toEqual(selected);
+        expect(batch_select_wrapper.vm.d_selected_items).not.toBe(selected);
+    });
+
+    test('closes the modal after cancelling', async () => {
+        expect(batch_select_wrapper.vm.d_show_batch_select_modal).toBe(false);
+
+        batch_select_wrapper.findAll('.batch-select-button').at(0).trigger('click');
+        await batch_select_wrapper.vm.$nextTick();
+
+        expect(batch_select_wrapper.vm.d_show_batch_select_modal).toBe(true);
+
+        batch_select_wrapper.findAll('.modal-cancel-button').at(0).trigger('click');
+        await batch_select_wrapper.vm.$nextTick();
```

Comment:
Are we missing an `expect(batch_select_wrapper.vm.d_show_batch_select_modal).toBe(false);` at the end here?

Reply:
Fixed

Reply:
For some reason I don't see the added `expect` anymore... Am I missing something?